pkg/reconciler/broker/broker.go

```diff
@@ -183,11 +191,41 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, b *eventingv1.Broker) pk
 
 	// Route everything to shared ingress, just tack on the namespace/name as path
 	// so we can route there appropriately.
-	b.Status.SetAddress(&apis.URL{
-		Scheme: "http",
-		Host:   network.GetServiceHostname(names.BrokerIngressName, system.Namespace()),
-		Path:   fmt.Sprintf("/%s/%s", b.Namespace, b.Name),
-	})
+	transportEncryptionFlags := feature.FromContext(ctx)
+	if transportEncryptionFlags.IsPermissiveTransportEncryption() {
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpAddress := r.httpAddress(b)
+		httpsAddress := r.httpsAddress(caCerts, b)
+		// Permissive mode:
+		// - status.address http address with host-based routing
+		// - status.addresses:
+		//   - https address with path-based routing
+		//   - http address with host-based routing
+		b.Status.Addresses = []pkgduckv1.Addressable{httpsAddress, httpAddress}
+		b.Status.Address = &httpAddress
+	} else if transportEncryptionFlags.IsStrictTransportEncryption() {
+		// Strict mode: (only https addresses)
+		// - status.address https address with path-based routing
+		// - status.addresses:
+		//   - https address with path-based routing
```

Comment:
Comments on path vs host based routing are not true, both addresses should use path-based routing, so we can remove the references to host-based routing